### PR TITLE
Rework bulk api, and add bulk api v2.0

### PR DESCRIPTION
**Pull request by morriswinkler-simple**
This changeset adds some code that is basically a cleanup of the bulk API, I also added Bulk API v2,0 which uses the REST API but supports only CSV.

### CODE REVIEW

File: force/bulkv2.go [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+package force
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+// MaxBatchv2SizeMiB is the maimum a batch can hold
```
Comment (a-urth) [inline force/bulkv2.go]: maximum

File: force/bulkv2.go [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+		if jobList.Done {
+			break
+		}
+		uri = strings.TrimPrefix(jobList.NextRecordsUrl, forceAPI.oauth.InstanceURL)
+	}
+
+	return
+}
+
+// AddBatch uploads a data batch of up to 150MB.
```
Comment (a-urth) [inline force/bulkv2.go]: there this check for 150mb is done?

File: force/bulk.go [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+
+	return &resp, nil
+}
+
+// AddBatch adds a new batch to a job by sending a POST request to the following URI.
+// The request body contains a list of records for processing.
+func (b *BulkJob) AddBatch(batch []byte) (*Batch, error) {
+	uri := fmt.Sprintf(`/services/async/%.1f/job/%s/batch`, b.APIVersion, b.ID)
+
+	//headers := http.Header{"Content-Type": []string{"text/csv"}}
```
Comment (a-urth) [inline force/bulk.go]: ?

File: force/bulkv2.go (hunk 2 above)
Comment (morriswinkler-simple) [reply force/bulkv2.go]: There is no check in the library. Only consts so you can check yourself.

**Comment by morriswinkler-simple** [comment]
I added an [example](https://github.com/simplesurance/go-force/blob/reworkBulkApi/force/example_bulk_test.go)

**Pull request merged**